### DISCUSSION TITLE
Alguém usando IndexedDB para SPA?

### DISCUSSION THREAD

**Discussion by juliovedovatto** [frontendbr/forum#1788] [Perguntas e Respostas]
No web app (TS, Vue, Vuex) que desenvolvo no trabalho, usamos Cookies para armazenar informações que precisam persistir pro usuário, como token de login, dados do usuário logado, dados do SPA (é app whitelabel), qual tema usuário selecionou na UI e outras coisas menores.

Logo precisaremos implementar features mais avançadas, como armazenar o tempo de vídeos que o usuário assistiu (como youtube/netflix faz). Dae fiquei ponderei e cheguei a  conclusão que Cookie não será a melhor escolha pra guardar e persistir informações assim.

Fiquei pensando se IndexedDB seria uma boa escolha pro caso, por causa de sua boa cobertura  (https://caniuse.com/#feat=indexeddb) e poder armazenar qualquer coisa, sem precisar parse de string.

Queira ver se alguém aqui na comunidade já fez um trabalho e obteu bons resultados. Pois mudar toda a lógica de persistência vai dar trabalho 😅

Obs: 

* Local Storage não foi considerado para este caso, pois alguns devs tiveram problemas com outro projeto, então nos requisitos do SPA que trabalho ele foi cortado fora.
* Tem gente na equipe que levanta a bandeira de aplicação cookieless (excerto para terceiros), então usar o IndexedDB seria uma boa resposta.
* Procurei aqui antes a respeito e não vi quase nada a respeito. Será que é tão fora da curva assim?

**Comment by kivervinicius** [comment]
cookie é a pior opção possível para estes dados, não entendo o motivo do localStorage ter sido "cortado fora" irá funcionar tanto quanto o indexedDb, eu já utilizo indexeddb com uma lib do firefox chamada localforage faz uns 7 anos e sempre com resultados excelentes.

**Comment by kivervinicius** [comment]
Referente ao uso no geral, praticamente todos app que conheço armazenam algo no indexedDb, exemplo o mega.co.nz  grava todo streaming de download no banco local, libs como firebase, reactivedb, pouchdb também utilizam a mesma estrutura para lançamento offline

**Comment by juliovedovatto** [comment]
@kivervinicius Justamente nas minhas pesquisas achei a localforage, achei bem promissora pelo fato de fazer wrapper do localStorage e IndexedDB.

Eu vi que players grandes usam IndexedDB, mas nunca conheci devs diretamente que faziam uso dela.

**Comment by kivervinicius** [comment]
@juliovedovatto vai com fé que é sucesso, a lib é muito boa, uso desde a versão beta para armazenar lançamentos offline e posterior sincronização, há muitos addons para ela que ajudam em questões rotineiras, por exemplo, https://github.com/localForage/localForage-observable

**Comment by ViniciusGularte** [comment]
Tava ajudando pessoal num desenvolvimento de um Pwa e estamos usando o [dexie](https://github.com/dfahlander/Dexie.js) como wrapper para o indexedDB,  ta funcionando muito bem para nosso caso.

**Comment by kivervinicius** [comment]
@ViniciusGularte o Dexie é muito bom realmente, pois permite trabalhar de forma mais elaborada com o indexedb porém o localforage gosto dele pela facilidade em inserir e obter os itens
